src/app/swap/swap-out.ts

```
@@ -0,0 +1,79 @@
+
+  const onChainBalance = await onChainService.getBalance()
+  if (onChainBalance instanceof Error) return onChainBalance
+  const outbound = await getChannelLiquidityBalance()
+  const minOnChainHotWalletBalanceConfig =
+    getColdStorageConfig().minOnChainHotWalletBalance
+  const minOutboundLiquidityBalance = getSwapConfig().minOutboundLiquidityBalance
+
+  const swapChecker = SwapOutChecker({
+    currentOnChainHotWalletBalance: onChainBalance,
+    minOnChainHotWalletBalanceConfig,
+    currentOutboundLiquidityBalance: outbound,
+    minOutboundLiquidityBalance,
+  })
+  const isOnChainWalletDepleted = swapChecker.isOnChainWalletDepleted()
+  const isOutboundLiquidityDepleted = swapChecker.isOutboundLiquidityDepleted()
+  logger.info(
+    { isOnChainWalletDepleted, isOutboundLiquidityDepleted },
+    "wallet depletion status",
+  )
+
+  addAttributesToCurrentSpan({
+    "swap.isOutboundLiquidityDepleted": isOutboundLiquidityDepleted,
+    "swap.isOnChainWalletDepleted": isOnChainWalletDepleted,
+  })
+
+  if (isOnChainWalletDepleted && isOutboundLiquidityDepleted) {
+    const swapResult = await swapService.swapOut(toSats(amount))
+    if (swapResult instanceof Error) {
+      addAttributesToCurrentSpan({
+        "swap.error": JSON.stringify(swapResult),
+      })
+    } else {
+      addAttributesToCurrentSpan({
+        "swap.submitted": JSON.stringify(swapResult),
+      })
+    }
+    return swapResult
+  }
+  return null // no swap needed
+}
+
+async function getChannelLiquidityBalance(): Promise<Satoshis | Error> {
+  try {
+    const activeNode = getActiveLnd()
+    if (activeNode instanceof Error) return activeNode
+    const lnd = activeNode.lnd
+    const { channel_balance, inbound } = await getChannelBalance({ lnd })
+    let outbound = 0
+    const inboundBal = inbound ? inbound : 0
+    if (inbound) {
+      outbound = channel_balance - inboundBal
+    }
+    return toSats(outbound)
+  } catch (err) {
+    return new Error(err)
```

app should return a specialized error (extended), not a generic one.